widgetssdk/src/main/java/com/glia/widgets/RequestCallback.kt

```diff
@@ -0,0 +1,24 @@
+package com.glia.widgets
+
+/**
+ * Callback used in some requests to Glia Widgets SDK.
+ *
+ * @param <T> object expected to be returned by the request
+</T> */
+interface RequestCallback<T> {
+
+    /**
+     * Function that is fired if request succeeds
+     *
+     * @param response result returned if the request succeeded
+     */
+    fun onSuccess(response: T?)
+
+    /**
+     * Function that is fired if request fails
+     *
+     * @param exception error returned if the request failed
+     */
+    fun onError(exception: GliaWidgetsException?)
```

Comment:
Why are the `response` and `exception` parameters are nullable?
Do we have cases where the successful response is empty?
If so I'd prefer having separate onSuccess Callback for thouse particular cases.

Reply:
I think we can have separate callbacks for success and error. For example
```kotlin
typealias OnSuccess<T> = (T) -> Unit
typealias OnError = (Throwable?) -> Unit
typealias OnSuccessCallback = () -> Unit
```

Reply:
@DavDo Maybe I'm missing something, but `typealias` name can directly be referenced only from Kotlin file, isn't it? We need it to work from Java classes as well. 

So I have created a separate `callbacks` package and put there three interfaces:
1. `OnWidgetsResult<T>`
2. `OnWidgetsSuccess`
3. `OnWidgetsError`

You can see how it looks like for get and update visitor info [here](https://github.com/salemove/android-sdk-widgets/pull/1257/files).

Looking forward to seeing your feedback.

Reply:
You're right!
TypeAliases are not directly accessible from java code, and if we pass it as a function parameter it will be converted to `Function0` or smth similar type, so having separate interfaces will look a way readable.